front-end/src/app/shared/utils/report-code.utils.ts

```diff
@@ -0,0 +1,227 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+export enum F3xReportCodes {
+  Q1 = 'Q1',
+  Q2 = 'Q2',
+  Q3 = 'Q3',
+  YE = 'YE',
+  TER = 'TER',
+  MY = 'MY',
+  TwelveG = '12G',
+  TwelveP = '12P',
+  TwelveR = '12R',
+  TwelveS = '12S',
+  TwelveC = '12C',
+  ThirtyG = '30G',
+  ThirtyR = '30R',
+  ThirtyS = '30S',
+  M2 = 'M2',
+  M3 = 'M3',
+  M4 = 'M4',
+  M5 = 'M5',
+  M6 = 'M6',
+  M7 = 'M7',
+  M8 = 'M8',
+  M9 = 'M9',
+  M10 = 'M10',
+  M11 = 'M11',
+  M12 = 'M12',
+}
+
+export class F3xReportCode {
+  code: string;
+  label: string;
+  coverageDatesFunction: ((year: number, isElectionYear: boolean, filingFrequency: string) => [Date, Date]) | undefined;
+
+  constructor(
+    code: string,
+    label: string,
+    coverageDatesFunction:
+      | ((year: number, isElectionYear: boolean, filingFrequency: string) => [Date, Date])
+      | undefined
+  ) {
+    this.code = code;
+    this.label = label;
+    this.coverageDatesFunction = coverageDatesFunction;
+  }
+}
+
+function createCoverageFunction(
+  startMonth: number,
+  startDate: number,
+  endMonth: number,
+  endDate: number
```

Comment:
A small nitpick: I think this would be more clear if startDate and endDate were renamed startDay and endDay. Initially, I saw the variable and assumed it was a Date() until I traced it back to the parameter in the function and saw it was a number

Reply:
I think it's confusing too.  I tried just matching what language [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date) uses.  For Date(), `Day` means 'Saturday', 'Friday', etc.  what if I called it `startDayOfMonth`?

Reply:
Per our discussion, I will go ahead and update with startDayOfMonth and endDayOfMonth